modules/margin/src/main/java/com/opengamma/sdk/margin/InvokerMarginClient.java

```diff
@@ -177,6 +180,67 @@ public MarginCalcResult calculate(Ccp ccp, MarginCalcRequest request) {
   }
 
   @Override
+  public MarginWhatIfCalcResult calculateWhatIf(Ccp ccp, MarginCalcRequest request, List<PortfolioDataFile> deltaFiles) {
+    String baseCalcId = createCalculation(ccp, request);
+    MarginCalcRequest secondRequest = MarginCalcRequest.builder()
```

Comment:
The files for the second request should be the files from the first request plus the delta. Also, use `toBuilder()`.

Use `toBuilder().portfolioData(combinedDelta).build();`